Application/Security.js

```diff
@@ -100,18 +100,117 @@ angular.module('shared')
           }
         );
       },
+      oAuth2Authorize: function(client_id, redirect_uri, response_type, scope, user, callback) {
+        var self = this;
+        
+        switch (response_type) {
```

Comment:
1. As was mentioned above ^ handling an undefined response_type would be good. I had a problem with this earlier and didn't know this was the cause because there was no error returned. 
2. I don't know if I am just missing something or what, but I don't see the point of having a callback if you are not using it. I get that the redirect uri is the callback on success, but wouldn't it make sense to return the errors in the callback?